Source/WebCore/platform/graphics/cg/ImageBufferIOSurfaceBackend.cpp

```diff
@@ -130,10 +129,10 @@ void ImageBufferIOSurfaceBackend::flushContext()
 
 bool ImageBufferIOSurfaceBackend::flushContextDraws()
 {
-    if (!m_context)
-        return false;
-    if (!m_context->consumeHasDrawn())
+    bool contextNeedsFlush = m_context && m_context->consumeHasDrawn();
+    if (!contextNeedsFlush && !m_needsFlush)
```

Comment:
If the clearRect() is moved to `ImageBufferIOSurfaceBackend::context()` then we do not need to change this function or to add `m_needsFlush`.

Reply:
Is it possible for a consumer to never access the context though?

I thought it'd be possible to have a caller just use `putPixelBuffer` (or `getPixelBuffer`), and we'd want the buffer to be cleared despite the `GraphicsContextCG` not yet existing.

Reply:
You are right.

But having already two variables about the dirtiness of the IOSurface 

```
ImageBufferIOSurfaceBackend::m_mayHaveOutstandingBackingStoreReferences
GraphicsContextCG::consumeHasDrawn()
```

makes me nerves for adding a third one. Can we avoid adding it? At least it should be renamed to `m_needsFirstFlush`, `m_needsFlushAfterCreation` or something else.

Reply:
I can't think of any way to avoid the extra variable unfortunately, but I've renamed it to `m_needsFirstFlush` to make the usage clearer.